### PR TITLE
Rithy/milestone1/entities

### PR DESCRIPTION
**Pull request by filipceko**
Base entities created, Reservation Entity and DAO implemented

### CODE REVIEW

File: TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/dao/ReservationDao.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package cz.muni.fi.travelAgency.dao;
+import cz.muni.fi.travelAgency.entities.Customer;
+import cz.muni.fi.travelAgency.entities.Excursion;
+import cz.muni.fi.travelAgency.entities.Reservation;
+import cz.muni.fi.travelAgency.entities.Trip;
+
+import java.util.Date;
+import java.util.List;
+
+public interface ReservationDao {
```
Comment (filipceko) [inline TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/dao/ReservationDao.java]: Inerface should also contain java doc for of the methods. Please add them in your next commit

File: TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/dao/ReservationDaoImp.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Override
+    public List<Reservation> findByCustomer(Customer customer) {
+        TypedQuery<Reservation> query = em.createQuery("SELECT e FROM Reservation e WHERE e.customer = :customerId",
+                Reservation.class);
+        query.setParameter("customerId", customer);
+        return query.getResultList();
+    }
+
+    public List<Reservation> findByTrip( Trip trip) {
```
Comment (filipceko) [inline TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/dao/ReservationDaoImp.java]: plese delete the obsolete space before Trip

File: TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package cz.muni.fi.travelAgency.entities;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.Date;
+import java.util.Objects;
+
+@Entity
+@Table(name = "RESERVATION")
+public class Reservation {
```
Comment (filipceko) [inline TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java]: Add Java doc comments please

File: TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    public Date getReserveDate() {
+        return reserveDate;
+    }
+
+    public void setReserveDate(Date reserveDate) {
+        this.reserveDate = reserveDate;
+    }
+
+
```
Comment (filipceko) [inline TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java]: Obsolete line, please delete.

File: TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Reservation)) return false;
+        Reservation that = (Reservation) o;
+        return Objects.equals(getCustomer(), that.getCustomer()) &&
```
Comment (simrau) [inline TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Reservation.java]: Reservation equals should be based on Customer and Trip, not on the day on which Reservation was created.

**Pull request merged**